lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -852,7 +852,11 @@ def move_directory_files( self, current_dir, source_dir, destination_dir ):
             os.makedirs(destination_directory)
         for dir_entry in os.listdir(source_directory):
             source_entry = os.path.join(source_directory, dir_entry)
-            if os.path.islink(source_entry):
+            if os.path.isdir(source_entry):
+                destination_subdir = os.path.join(destination_directory, dir_entry)
+                if not os.path.exists(destination_subdir):
+                    os.makedirs(destination_subdir)
+            elif os.path.islink(source_entry):
```

Comment:
Yes, I see what you mean now - thanks!

I've played around with this a bit more, and the current version of this function does not correctly move subdirectories.  If the directory being moved ioncludes a subdirectory, the following error can be thrown:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 13, in move_directory_files
  File "/usr/lib64/python2.7/shutil.py", line 291, in move
    raise Error, "Destination path '%s' already exists" % real_dst
shutil.Error: Destination path '/tmp/testdir/subdir' already exists
```

However, the original version of the function shown here worked for all content types.  
```
    def move_directory_files( self, current_dir, source_dir, destination_dir ):
        source_directory = os.path.abspath( os.path.join( current_dir, source_dir ) )
        destination_directory = os.path.join( destination_dir )
        if not os.path.isdir( destination_directory ):
            os.makedirs( destination_directory )
        symlinks = []
        regular_files = []
        for file_name in os.listdir( source_directory ):
            source_file = os.path.join( source_directory, file_name )
            destination_file = os.path.join( destination_directory, file_name )
            files_tuple = ( source_file, destination_file )
            if os.path.islink( source_file ):
                symlinks.append( files_tuple )
            else:
                regular_files.append( files_tuple )
        for source_file, destination_file in symlinks:
            shutil.move( source_file, destination_file )
        for source_file, destination_file in regular_files:
            shutil.move( source_file, destination_file )
```
Should we fix the existing version or revert back to the original?